### ISSUE TITLE
module throws for ?? reasons

### ISSUE THREAD

**Issue by sandfox** [yeller/yeller_node#1]
Not entirely sure whats happening here, but so far it looks like our application is going bang, and then yeller is going bang too when we report it...

_more info_

The original error is being caused by some malformed JSON file

```
[2015-04-24T15:04:07.675Z]  WARN: exception-tracker/16876 on ip-10-72-139-28: yeller error [TypeError: Cannot call method 'lastIndexOf' of null]
[2015-04-24T15:04:07.677Z]  WARN: exception-tracker/16876 on ip-10-72-139-28: actual error [SyntaxError: /var/www/project-x/libs/payment/bannedDevicesList.json: Unexpected string]
```

``` js
/var/www/project-x/node_modules/yeller_node/index.js:9
    if (project_root && frame.fileName.lastIndexOf(project_root, 0) === 0 && f
                                       ^
TypeError: Cannot call method 'lastIndexOf' of null
  at formatFrames (/var/www/project-x/node_modules/yeller_node/index.js:9:40)
  at formatError (/var/www/project-x/node_modules/yeller_node/index.js:22:19)
  at [object Object].YellerClient.formatJSONError (/var/www/project-x/node_modules/yeller_node/index.js:61:19)
  at [object Object].YellerClient.report (/var/www/project-x/node_modules/yeller_node/index.js:118:19)
  at tell_ya (/var/www/project-x/libs/exception/index.js:36:23)
  at /var/www/project-x/node_modules/async/lib/async.js:570:21
  at /var/www/project-x/node_modules/async/lib/async.js:249:17
  at /var/www/project-x/node_modules/async/lib/async.js:125:13
  at Array.forEach (native)
  at _each (/var/www/project-x/node_modules/async/lib/async.js:46:24)
  at async.each (/var/www/project-x/node_modules/async/lib/async.js:124:9)
  at _asyncMap (/var/www/project-x/node_modules/async/lib/async.js:248:13)
  at Object.map (/var/www/project-x/node_modules/async/lib/async.js:219:23)
  at _parallel (/var/www/project-x/node_modules/async/lib/async.js:568:20)
  at Object.async.parallel (/var/www/project-x/node_modules/async/lib/async.js:598:9)
  at track (/var/www/project-x/libs/exception/index.js:41:11)
  at process.<anonymous> (/var/www/project-x/libs/shutdown/index.js:59:9)
  at process.g (events.js:180:16)
  at process.emit (events.js:95:17)
  at process._fatalException (node.js:295:26)
  at process._asyncFatalException [as _fatalException] (/var/www/project-x/node_modules/async-listener/glue.js:211:34)
```

**Comment by sandfox** [comment]
so yeah, found the problem...

**TL:DR**

I'm not sure the way you want to handle it but basically it looks like you need to defend against `fileName`, `lineNumber` and `columnNumber` inside a `frame` being `null`, at a glance maybe you could check for `native === true` on the frame?

_longer explanation_

If for example we try to throw some invalid JSON through `json.parse` and that exception gets thrown into the `yeller`module. the yeller module chucks the `error` through `stackparse`, and that returns something like the following (example taken from some poc code i wrote)

``` js
// actual error message
[SyntaxError: /Users/sandfox/tmp/yeller-boom/bad.json: Unexpected token a]

//stackparse output
[ { fileName: null,
    lineNumber: null,
    functionName: 'Object.parse',
    typeName: 'Object',
    methodName: 'parse',
    columnNumber: null,
    native: true },
  { fileName: 'module.js',
    lineNumber: 454,
    functionName: 'Object.Module._extensions..json',
    typeName: 'Object',
    methodName: 'Module._extensions..json',
    columnNumber: 27,
    native: false },
  { fileName: 'module.js',
    lineNumber: 353,
    functionName: 'Module.load',
    typeName: 'Module',
    methodName: 'load',
    columnNumber: 32,
    native: false },
  { fileName: 'module.js',
    lineNumber: 308,
    functionName: 'Function.Module._load',
    typeName: 'Function',
    methodName: 'Module._load',
    columnNumber: 12,
    native: false },
  { fileName: 'module.js',
    lineNumber: 363,
    functionName: 'Module.require',
    typeName: 'Module',
    methodName: 'require',
    columnNumber: 17,
    native: false },
  { fileName: 'module.js',
    lineNumber: 382,
    functionName: 'require',
    typeName: 'Object',
    methodName: null,
    columnNumber: 17,
    native: false },
  { fileName: '/Users/sandfox/tmp/yeller-boom/index.js',
    lineNumber: 35,
    functionName: '',
    typeName: 'Object',
    methodName: null,
    columnNumber: 1,
    native: false },
  { fileName: 'module.js',
    lineNumber: 428,
    functionName: 'Module._compile',
    typeName: 'Module',
    methodName: '_compile',
    columnNumber: 26,
    native: false },
  { fileName: 'module.js',
    lineNumber: 446,
    functionName: 'Object.Module._extensions..js',
    typeName: 'Object',
    methodName: 'Module._extensions..js',
    columnNumber: 10,
    native: false },
  { fileName: 'module.js',
    lineNumber: 353,
    functionName: 'Module.load',
    typeName: 'Module',
    methodName: 'load',
    columnNumber: 32,
    native: false } ]
```

The problem starts when we try to do stuff with `frame.fileName` on that frame at the top of that array of frames. I'm not sure why there is no filename for that frame (maybe it's a function or native code from the depths of V8?).

**Comment by tom-pang** [comment]
Fixed in https://github.com/yeller/yeller_node/commit/05e04cfce0fbc3a90800aa1de95a1ee99c36f8a1 and released as `0.0.7`. Let me know how it goes!

**Comment by sandfox** [comment]
Works for me, Awesome, thanks!

**Comment by tom-pang** [comment]
awesome. Sorry you hit this.

Let me know if there's anything else I can do for you—I'm happy to help
